pycalphad/tests/test_utils.py

```diff
@@ -15,9 +15,27 @@ def test_filter_phases_removes_disordered_phases_from_order_disorder():
     filtered_phases = set(filter_phases(ALNIPT_DBF, unpack_components(ALNIPT_DBF, ['AL', 'NI', 'PT', 'VA'])))
     assert all_phases.difference(filtered_phases) == {'FCC_A1'}
 
+
 def test_filter_phases_removes_phases_with_inactive_sublattices():
     """Phases that have no active components in any sublattice should be filtered"""
     all_phases = set(ALNIPT_DBF.phases.keys())
     filtered_phases = set(filter_phases(ALNIPT_DBF, unpack_components(ALNIPT_DBF, ['AL', 'NI', 'VA'])))
     assert all_phases.difference(filtered_phases) == {'FCC_A1', 'PT8AL21', 'PT5AL21', 'PT2AL', 'PT2AL3', 'PT5AL3', 'ALPT2'}
 
+
+def test_instantiate_models_only_returns_desired_phases():
+    """instantiate_models should only return phases passed"""
+    comps = ['AL', 'NI', 'VA']
+    phases = ['FCC_A1', 'LIQUID']
+
+    # models are overspecified w.r.t. phases
+    too_many_phases = ['FCC_A1', 'LIQUID', 'AL3NI1']
+    too_many_models = {phase: Model(ALNIPT_DBF, comps, phase) for phase in too_many_phases}
+    inst_mods = instantiate_models(ALNIPT_DBF, comps, phases, model=too_many_models)
+    assert len(inst_mods) == len(phases)
+
+    # models are underspecified w.r.t. phases
+    too_few_phases = ['FCC_A1', 'LIQUID', 'AL3NI1']
```

Comment:
Is this test correct? It looks identical to the one above it.

Reply:
Good catch, should be fixed